### ISSUE TITLE
Email Providers: Improve how we display the pricing information

### ISSUE THREAD

**Issue by taipeicoder** [Automattic/wp-calypso#95929]
### Quick summary

The pricing information is hard to parse. See screenshot for reference:

![Image](https://github.com/user-attachments/assets/a8477772-3613-4190-88bc-b448a3d5b46e)

In this example, it says 3 months free, but also $0/year which doesn't seem accurate. It's hard for users to understand what happens/how much they will be charged after the trial.

Also, having "email" listed in what's included is redundant.

### Steps to reproduce

1. Start at /domains/manage/:domain
2. Click on "Add new domain"
3. Select any domain
4. On the email providers screen, there will be the pricing displayed.

### What you expected to happen

Pricing information displayed in a clear way.

### What actually happened

Pricing information is confusing.

### Impact

One

### Available workarounds?

There is no user impact

**Comment by supernovia** [comment]
📌 ACTIONS
- Triaged
- Enhancement kept

I'd love to see us just charge $35 after the end of the trial, and have that go one year. :)
